### PR TITLE
Refactor extract LoginOfflineLink to 'gui.auth.use_offline' namespace

### PR DESCRIPTION
**Pull request by gonzalo-bulnes**
# Description

This is the 6th step to be split out of #1369.
It moves the `LoginOfflineLink` definition to its own file/module. Eventually, the `LoginDialog` will be part o the `auth` package too,.

This step is necessary to prevent a circular dependency when extracting the `LoginDialog` out of `widgets.py`.

# Test Plan

- [ ] Confirm that the "Use Offline" (in the login dialog) behavior and appearance haven't changed
- [ ] Confirm that the `LoginOfflineLink` definition is unchanged
- [ ] Confirm that the `LoginOfflineLink` didn't have any unit test
- [ ] Confirm that CI is green

# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:

 - [ ] I have tested these changes in the appropriate Qubes environment
 - [ ] I do not have an appropriate Qubes OS workstation set up (the reviewer will need to test these changes)
 - [x] These changes should not need testing in Qubes

If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:

 - [ ] I have updated the [AppArmor profile](https://github.com/freedomofpress/securedrop-client/blob/HEAD/files/usr.bin.securedrop-client)
 - [x] No update to the AppArmor profile is required for these changes
 - [ ] I don't know and would appreciate guidance

If these changes modify the database schema, you should include a database migration. Please check as applicable:

 - [ ] I have written a migration and upgraded a test database based on `main` and confirmed that the migration applies cleanly
 - [ ] I have written a migration but have not upgraded a test database based on `main` and would like the reviewer to do so
 - [ ] I need help writing a database migration
 - [x] No database schema changes are needed

### CODE REVIEW

File: securedrop_client/gui/login_dialog.py [hunk 1]
```diff
@@ -35,12 +35,8 @@
 )
 
 from securedrop_client import __version__ as sd_version
-from securedrop_client.gui.widgets import (
-    LoginErrorBar,
-    LoginOfflineLink,
-    PasswordEdit,
-    SignInButton,
-)
+from securedrop_client.gui.auth import LoginOfflineLink
```
Comment (gonzalo-bulnes) [inline securedrop_client/gui/login_dialog.py:38]: As previously (see https://github.com/freedomofpress/securedrop-client/pull/1379/files#r771878542) here is agin the clue that `LoginDialog` belongs to `auth`.

Comment (gonzalo-bulnes) [reply securedrop_client/gui/login_dialog.py:38]: Will become the following once #1379 and #1380 are merged:
```diff
- from securedrop_client.gui.auth import LoginErrorBar
+ from securedrop_client.gui.auth import LoginErrorBar, LoginOfflineLink, SignInButton
```

File: securedrop_client/gui/login_dialog.py [hunk 2]
```diff
@@ -35,12 +35,8 @@
...
+from securedrop_client.gui.widgets import LoginErrorBar, PasswordEdit, SignInButton
```
Comment (gonzalo-bulnes) [inline securedrop_client/gui/login_dialog.py:39]: Will become the following after #1378, #1379 and #1380 are merged:
```diff
- from securedrop_client.gui.widgets import LoginOfflineLink, PasswordEdit, SignInButton
+ from securedrop_client.gui.base import PasswordEdit
```

File: securedrop_client/locale/messages.pot [hunk 3]
```diff
@@ -313,3 +310,6 @@ msgstr ""
 msgid "Files and messages"
 msgstr ""
 
+msgid "USE OFFLINE"
+msgstr ""
+
```
Comment (conorsch) [inline securedrop_client/locale/messages.pot:315]: These `.pot` changes are required due to the refactor. During review, @gonzalo-bulnes mentioned that CI was smart enough to fail here pending regeneration of this file. So hat-tip to @cfm on the CI logic, making sure this file is up to date!

File: securedrop_client/gui/auth/__init__.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 """
 Widgets related to authentication.
 
-Copyright (C) 2018  The Freedom of the Press Foundation.
+Copyright (C) 2021  The Freedom of the Press Foundation.
```
Comment (conorsch) [inline securedrop_client/gui/auth/__init__.py:4]: Thanks for updating the copyright date! We'll want to do this throughout in the near future, but certainly not worth blocking on in the meantime.

**Review by conorsch — APPROVED** [review]
:+1:

**Pull request merged**